File: 프로그래머스/2/42842. 카펫/카펫.js
function solution(brown, yellow) {
    var answer = [];
    let total= brown + yellow
    for(let row=1;row<=brown;row++){
        for(let col=1;col<=row;col++){
            if(row*col===total&&(row-2)*(col-2)===yellow){
                answer.push(row)
                answer.push(col)
            }
        }
    }
    return answer;
}

File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    answer = []
    for i in range(1, brown+1):
        for j in range(1, brown+1):
            if(i*j==brown+yellow and (j-2)*(i-2)== yellow and (j*2+2*i-4)==brown):
                return [j,i]